pkgs/development/interpreters/python/hooks/default.nix

```diff
@@ -189,6 +189,19 @@ in {
       };
     } ./python-runtime-deps-check-hook.sh) {};
 
+  relaxBuildSystemRequiresHook = callPackage ({ makePythonHook, packaging, tomlkit }:
+    makePythonHook {
+      name = "relax-build-system-requires-hook";
+      propagatedBuildInputs = [ packaging tomlkit ];
```

Comment:
FYI, tomlkit in turn depends on pyyaml, pytest and poetry-core, which now risks being load-bearing for the majority of python3Packages. This is far better than introducing tomlq or dasel. :+1:

Reply:
Could we maybe use the inbuilt tomli and tomli-w? THen we only need poetry-core. For pytest we could disable tests.

Reply:
I feel it is way more safe to use the tool that preserves whitespace and comments

Reply:
tomlkit is recommended for editing existing toml.
Also, poetry-core is the only dependency when tests are disabled.

> The [TOML Kit package](https://pypi.org/project/tomlkit/) is a style-preserving TOML library with both read and write capability. It is a recommended replacement for this module for editing already existing TOML files.

https://docs.python.org/3/library/tomllib.html